src/main/java/org/opensearch/security/auth/limiting/AbstractRateLimiter.java

```diff
@@ -35,6 +39,8 @@ public abstract class AbstractRateLimiter<ClientIdType> implements AuthFailureLi
     protected final RateTracker<ClientIdType> rateTracker;
     protected final List<String> ignoreHosts;
     private WildcardMatcher ignoreHostMatcher;
+    // Cache for subnet matchers
+    private static final Map<String, SubnetUtils.SubnetInfo> SUBNET_CACHE = new ConcurrentHashMap<>();
```

Comment:
Given that the `auth_failure_listeners` are part of the dynamic config, we will need to ensure this cache is updated on a config change to the `ignore_hosts` setting for completeness. I think it would be ok to address in a follow-up, but we should capture it in an issue if its not already being handled.

See the `@Subscribe` section of [BackendRegistry](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/auth/BackendRegistry.java#L176-L194) for an example of how to listen to changes in the dynamic configuration.

Reply:
Yes, while the current implementation handles additions, removals are not handled. i.e. since there's no TTL - entries are persisted for lifetime of the node. Let me check if I can switch this with a TTL based cache or Subscribe to changes. (could also be a follow-up PR)